lib/task/import/validate/csvColumnCountValidator.class.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * CSV column count test. Test all rows in CSV have the same number of columns.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ *
+ * @internal
+ * @coversNothing
+ */
+class CsvColumnCountValidator extends CsvBaseValidator
+{
+    const TITLE = 'CSV Column Count Check';
+    protected $headerCount;
+    protected $rowCountSummary = [];
+
+    public function __construct(array $options = null)
+    {
+        $this->setTitle(self::TITLE);
+
+        parent::__construct($options);
+    }
+
+    public function reset()
+    {
+        $this->headerCount = null;
+        $this->rowCountSummary = [];
+
+        parent::reset();
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        parent::testRow($header, $row);
+
+        if (!isset($this->headerCount)) {
+            $this->headerCount = count($header);
+
+            $this->updateRowCountSummary($this->headerCount);
+        }
+
+        $this->updateRowCountSummary(count($row));
+    }
+
+    public function getTestResult()
+    {
+        // When rows are all same length then rowCountSummary will have 1 row.
+        if (1 == count($this->rowCountSummary)) {
```

Comment:
Cosmetic: "Always use identical comparison unless you need type juggling;" -- https://symfony.com/doc/current/contributing/code/standards.html

Reply:
👍

Reply:
Addressed in mulitple places...